lib/versioning/pvp/index.ts

```diff
@@ -0,0 +1,165 @@
+import { VersioningApi } from '../types';
+
+export const id = 'pvp';
+export const displayName = 'PVP';
+export const urls = ['https://pvp.haskell.org'];
+export const supportsRanges = true;
+
+// https://pvp.haskell.org/#version-numbers
+type PVP = {
```

Comment:
You should use our generic versioning class as base for this, as it will provide most of your implemention.

Reply:
Hi, thanks for reviewing! Could you point out the generic versioning class? When I look at the other implementations I don't see any of them using a generic versioning class.

Ruby for instance takes a similar approach:
```
interface RubyVersion {
  major: number;
  minor: number;
  patch: number;
  prerelease: string[] | null;
}
```

Other ones seem to use the npm implementation if they use semver or pep440.


helm:
```
export const api: VersioningApi = {
  ...npm,
}
```

rez:
```
function getMajor(version: string): number {
  try {
    return npm.getMajor(padZeroes(version));
  } catch (err) /* istanbul ignore next */ {
    return pep440.getMajor(version);
}
```

Note those other implementations would be incompatible with PVP.

Reply:
Maybe @viceice meant using this https://github.com/renovatebot/renovate/blob/ee79cf7975a8cdda4952e38dc4d7d5676e350ec2/lib/versioning/loose/generic.ts ?

Reply:
yes, this:

https://github.com/renovatebot/renovate/blob/68dfc27f3981fda948330e919aac986bdf2a0229/lib/versioning/loose/generic.ts#L131

the stuff before is obsolete and will be removed soon

Reply:
Thanks for pointing that out. I'll try to refactor with that class now. That file also seems to calify our questions on ranges: https://github.com/renovatebot/renovate/blob/ee79cf7975a8cdda4952e38dc4d7d5676e350ec2/lib/versioning/loose/generic.ts#L18-L20

Reply:
I have updated my PR to use the Generic versioning class now.